torch/csrc/jit/script/compiler.cpp

```diff
@@ -1358,33 +1258,23 @@ struct to_ir {
 
       emitStatements(body);
 
-      // Also emit the conditional
-      cond_val = (cond) ? emitCond(cond.value())
-                        : graph->insertConstant(true, nullptr, range);
-      body_block->registerOutput(cond_val);
-      auto body_frame = popFrame();
-      auto outer_frame = environment_stack;
-
-      // Add block outputs to correspond to each captured input
-      // some of these will be removed.
-      for (const auto& x : body_frame->captured_inputs) {
-        auto fv = body_frame->getValueInThisFrame(range, x);
-        body_block->registerOutput(fv);
+      Node* loop_condition =
```

Comment:
This is really confusing. There is no comment anywhere describing what this structure is. Why is LoopCondition a separate node? Why does Loop not just have two blocks like you would expect?

Reply:
Why does the loop block have inputs and outputs still?